### PR TITLE
vfprintf: change long double to double 

### PR DESCRIPTION
**Pull request by palainp**
I'm not sure this is the best way to fix issue #110 (and as a side effect also fix mirage/qubes-mirage-firewall#134). Basically, it's a trade for the `double long` precision into `double`, which may or may not have an impact when we want to print floating values.

Since the initial problem doesn't seem to be seen outside of Qubes 4.1 (Xen 4.14) and can be caused by something other than that (maybe any change in the defaults between Qubes 4.0.4 and Qubes 4.1 (Xen 4.8 and Xen 4.14 respectively)) and we may have another floating point issue in another piece of code, I'm definitely not sure that lowering the precision here is the way to go  (it's probably better to fix the root cause but I'm now out of ideas for that) but it does have the advantage of allowing unikernel mirage to run with Qubes :)

### CODE REVIEW

**Comment by palainp** [comment]
Thinking back about the precision drop, and according to https://ocaml.org/manual/values.html, the floating point values are 64bits long (sign bit+53 mantissa bits+10 exponent bits) which will fits into `double` and doesn't seem to require `long double` (the original openlibm may have targets that will take the 16B for `long double` but I think that it's fine here).

Also when I try to force a long double precision print the compiler complains about the print format:
```
File "unikernel.ml", line 10, characters 42-45:
10 |         Logs.info (fun f -> f "hello %Lf" 0.3);
                                               ^^^
Error: This expression has type float but an expression was expected of type
         int
```

**Comment by hannesm** [comment]
> However, with a deep look, I can not find a call of vfprintf in the caml runtime.

In ocaml-freestanding, `vfprintf` is the actual implementation and all derivates (`fprintf`, `printf`, `vsnprintf`) are calling into `vfprintf`.

The OCaml runtime is slightly complex in respect to printf, but on the OCaml side:
- CamlInternalFormat defines a recursive function `make_printf` <- this is called by `Printf.printf`
- This calls out to `make_float_padding_precision`, which calls `convert_float`, which calls `format_float` which is `caml_format_float`
- That is a C implementation in `runtime/floats.c`, which calls `caml_alloc_sprintf` (in `runtime/str.c`), which calls out to `vsnprintf`

I think this PR is nice to merge, though I'm still unsure what happened between Xen 4.8 and Xen 4.14 and whether we'll run in other issues (e.g. with openlibm and trigonometry functions) that are fixed by float/double. I also don't have much knowledge about floating point numbers and precision.

TL;DR: Happy to have this merged since it fixes a concrete issue on Xen.

**Comment by dinosaure** [comment]
Ok, I think I understand more the execution path with what @hannesm said. In that case, I think that it's good to merge too :+1:. Thanks for your work.
